javascript/react/redux-tips/async-flow-in-redux.md

```diff
@@ -30,31 +31,65 @@ The default **Redux** store only supports synchronous data flow. However, with t
 
 Packages such as `redux-thunk` and `redux-promise` provide syntax sugar as to help you achieve such data flow.
 
-The main difference between a custom implementation of middleware or using a specific package is that the former wraps the store’s `dispatch()` method and allows you to dispatch something other than actions such as functions. This functions do not have to be **pure** so asynchronous API calls are permitted.
-
-Code sample for using `redux-thunk`:
-
-```javascript
-// action creator
-function loadData(userId) {
-  return dispatch => fetch(
-`http://data.com/${userId}`)
-// Redux Thunk handles these
-    .then(res => res.json())
-    .then(
-      data => dispatch({
-type: 'LOAD_DATA_SUCCESS', data }),
-      err => dispatch({
-type: 'LOAD_DATA_FAILURE', err })
-    );
+The main difference between a custom implementation of middleware or using a specific package is that the former wraps the store’s `dispatch()` method and allows you to dispatch something other than actions such as functions. This functions do not have to be **pure**, so asynchronous API calls are permitted.
```

Comment:
I'd remove this entire paragraph, it doesn't flow with the rest of the content

Reply:
Alternatively, add some information on what a middleware is and how it's used.

[Good start](https://read.reduxbook.com/markdown/part1/05-middleware-and-enhancers.html)

Reply:
Hmm seeing that the rest of the content bellow uses middlewares, we should probably explain a bit better what they are. 

I'd replace this whole paragraph by describing something like the following.

> There are 2 ways of adding capabilities to Redux, whether that is handling async operations or something else. The first one is to add one or more middlewares. The second is by enhancing the Redux store. <add a bit more info here>